### PR TITLE
Slow control loop with dwa_local_planner (fix for #679)

### PR DESCRIPTION
**Pull request by pavloblindnology**
Fix for #679

### CODE REVIEW

**Comment by mikeferguson** [comment]
CI didn't trigger initially -- now running

File: base_local_planner/src/goal_functions.cpp [hunk 1]
```diff
@@ -103,12 +104,8 @@ namespace base_local_planner {
     try {
       // get plan_to_global_transform from plan frame to global_frame
       tf::StampedTransform plan_to_global_transform;
-      tf.waitForTransform(global_frame, ros::Time::now(),
-                          plan_pose.header.frame_id, plan_pose.header.stamp,
-                          plan_pose.header.frame_id, ros::Duration(0.5));
-      tf.lookupTransform(global_frame, ros::Time(),
```
Comment (mikeferguson) [inline base_local_planner/src/goal_functions.cpp:109]: I wonder if would be desirable to keep this as ros::Time() which will get the latest available transform (rather than a possibly outdated one). Although, it might also be that having all transforms done at the same time is better for consistency.

File: base_local_planner/include/base_local_planner/goal_functions.h [hunk 2]
```diff
@@ -117,6 +118,7 @@ namespace base_local_planner {
   bool getGoalPose(const tf::TransformListener& tf,
   		  const std::vector<geometry_msgs::PoseStamped>& global_plan,
   		  const std::string& global_frame,
+        const ros::Time& time,
```
Comment (mikeferguson) [inline base_local_planner/include/base_local_planner/goal_functions.h]: The indentation here is off -- but it appears that is because you properly used spaces, but the lines above/below use some tabs.

File: dwa_local_planner/src/dwa_planner_ros.cpp [hunk 3]
```diff
@@ -251,10 +260,6 @@ namespace dwa_local_planner {
 
   bool DWAPlannerROS::computeVelocityCommands(geometry_msgs::Twist& cmd_vel) {
     // dispatches to either dwa sampling control or stop and rotate control, depending on whether we have been close enough to goal
-    if ( ! costmap_ros_->getRobotPose(current_pose_)) {
-      ROS_ERROR("Could not get robot pose");
-      return false;
-    }
```
Comment (mikeferguson) [inline dwa_local_planner/src/dwa_planner_ros.cpp:259]: why is this removed?

File: base_local_planner/include/base_local_planner/goal_functions.h (hunk 2 above)
Comment (pavloblindnology) [reply base_local_planner/include/base_local_planner/goal_functions.h]: Fixed.

File: base_local_planner/src/goal_functions.cpp (hunk 1 above)
Comment (pavloblindnology) [reply base_local_planner/src/goal_functions.cpp:109]: We cannot just replace `ros::Time::now()` with `ros::Time()` in line
https://github.com/ros-planning/navigation/blob/43e9297d588276efa9868efe2ed56ac6bf08e180/base_local_planner/src/goal_functions.cpp#L106
as transform in line
https://github.com/ros-planning/navigation/blob/43e9297d588276efa9868efe2ed56ac6bf08e180/base_local_planner/src/goal_functions.cpp#L115
will use possibly later timestamp of current robot pose, issuing TF error 'extrapolation into the future', mentioned here #196

File: dwa_local_planner/src/dwa_planner_ros.cpp (hunk 3 above)
Comment (pavloblindnology) [reply dwa_local_planner/src/dwa_planner_ros.cpp:259]: It's explained in line 157 of my dwa_planner_ros.cpp.
If we update robot pose after setting time in planner_util we can get a more recent time stamp of robot pose, which will cause TF 'extrapolation into the future` error in line
https://github.com/ros-planning/navigation/blob/43e9297d588276efa9868efe2ed56ac6bf08e180/base_local_planner/src/goal_functions.cpp#L115

**Comment by pavloblindnology** [comment]
Any update on this?

**Comment by mikeferguson** [comment]
Targets a dead release (kinetic). If this is still of interest, please open a PR to noetic-devel branch
